### PR TITLE
Add TableDistanceMatrix filter

### PR DESCRIPTION
**Pull request by pierre-guillou**
This PR adds a new VTK module, `ttkTableDistanceMatrix`, that calls the `lDistanceMatrix` base module on vtkTables.
The corresponding feature in the `RipsComplex` module (computing a distance matrix from coordinates in a high-dimension space) has been removed. Now, `RipsComplex` only acts on distance matrices (+ a lower dimension embedding).

Enjoy,
Pierre

### CODE REVIEW

File: core/vtk/ttkTableDistanceMatrix/ttkTableDistanceMatrix.cpp [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    arrays.push_back(input->GetColumnByName(s.data()));
+  }
+
+  std::vector<std::vector<double>> inputMatrix(numberOfRows);
+#ifdef TTK_ENABLE_OPENMP
+#pragma omp parallel for num_threads(this->threadNumber_)
+#endif // TTK_ENABLE_OPENMP
+  for(int i = 0; i < numberOfRows; ++i) {
+    for(size_t j = 0; j < numberOfColumns; ++j) {
+      inputMatrix[i].emplace_back(arrays[j]->GetVariantValue(i).ToDouble());
```
Comment (julien-tierny) [inline core/vtk/ttkTableDistanceMatrix/ttkTableDistanceMatrix.cpp:81]: Note for later @pierre-guillou 
anyway to pass pointers to the raw data down to the base code (like we do for the triangulation?)
(i.e. without input data copy)

Comment (pierre-guillou) [reply core/vtk/ttkTableDistanceMatrix/ttkTableDistanceMatrix.cpp:81]: The `lDistanceMatrix`/`lDistance` modules work on coordinates pointers: one pointer for each point, but the vtkTable arrays are accessed by column. This copy of the input data also serves to transpose the coordinates array to feed it to base modules (while taking care of typing by converting everything to double).
The same thing happens in `ttkDimensionReduction` (from which this snippet is taken).

Comment (julien-tierny) [reply core/vtk/ttkTableDistanceMatrix/ttkTableDistanceMatrix.cpp:81]: Makes sense, thanks!

**Comment by julien-tierny** [comment]
ok, let's go.

**Pull request merged**